src/main/java/com/target/ready/library/system/Controller/LibraryController.java

```diff
@@ -18,10 +18,20 @@ public class LibraryController {
     LibraryController(LibrarySystemService librarySystemService){
         this.librarySystemService=librarySystemService;
     }
-    @GetMapping("getBooks")
-    public List<Book> getAllBooks(){
-
-        return librarySystemService.getAllBooks();
+    @GetMapping("books_directory")
+    public List<Book> getAllBooks(@RequestParam(value = "pageNumber", defaultValue = "0", required = false) Integer pageNumber){
+        List<Book> books;
+        int pageSize=5;
+        System.out.println("page num:"+pageNumber);
+        try {
+            if (pageNumber < 0) {
+                return null;
```

Comment:
What will be the UI behaviour if the pageNumber < 0 ?

Reply:
The page will not be changed and remain in the same page and same with the case when the page Number exceeds the number of available pages